sphinx/ext/intersphinx.py

```diff
@@ -570,34 +574,49 @@ def get_inventory_and_name_suffix(self, name: str) -> tuple[str | None, str]:
             raise ValueError(msg)
 
     def get_role_name(self, name: str) -> tuple[str, str] | None:
+        """Given a string like ``domain:name`` or ``name``,
+        return a tuple of the (domain, role name),
+        or None if no available domain/role can be found.
+
+        The ``name`` can be either a role name or an object name.
+        """
         names = name.split(':')
         if len(names) == 1:
-            # role
             default_domain = self.env.temp_data.get('default_domain')
             domain = default_domain.name if default_domain else None
-            role = names[0]
+            name = names[0]
         elif len(names) == 2:
-            # domain:role:
             domain = names[0]
-            role = names[1]
+            name = names[1]
         else:
             return None
 
-        if domain and self.is_existent_role(domain, role):
+        if domain and (role := self.is_existent_role(domain, name)):
             return (domain, role)
-        elif self.is_existent_role('std', role):
+        elif (role := self.is_existent_role('std', name)):
             return ('std', role)
         else:
             return None
 
-    def is_existent_role(self, domain_name: str, role_name: str) -> bool:
+    def is_existent_role(self, domain_name: str, role_or_obj_name: str) -> None | str:
```

Comment:
I don't like that a predicate-like named method does not return a boolean. To be consistent, I would suggest having another method which resolves the role or object name and use that method in `is_existent_role` by returning whether it is None or not.

Reply:
done